In [1]:
import pandas as pd

In [2]:
airbnb = pd.read_csv('airbnb_data.csv')

In [3]:
%%sql
SELECT 
    CORR(price, rating) AS correlation_price_rating
FROM 
    airbnb
WHERE 
    price IS NOT NULL
    AND rating IS NOT NULL;

correlation_price_rating
0                 -0.004777

In [4]:
%%sql
SELECT 
    CORR(availability_365, number_of_reviews) AS correlation_availability_reviews
FROM 
    airbnb
WHERE 
    availability_365 IS NOT NULL
    AND number_of_reviews IS NOT NULL;

correlation_availability_reviews
0                          0.188975

In [5]:
%%sql
SELECT 
    CORR(number_of_reviews, rating) AS correlation_reviews_rating
FROM 
    airbnb
WHERE 
    number_of_reviews IS NOT NULL
    AND rating IS NOT NULL;

correlation_reviews_rating
0                   -0.002888

Correlation between Price and Ratings:

- Calculated the correlation coefficient between the price and ratings of Airbnb listings.
- Computed the correlation coefficient, filtering out null values for price or rating.
- Result: The correlation coefficient between price and ratings is approximately -0.004777, indicating almost no linear relationship between these variables.

Correlation between Availability and Number of Reviews:

- Examined the correlation between the availability and the number of reviews of Airbnb listings.
- Computed the correlation coefficient, filtering out null values for availability or number of reviews.
- Result: The correlation coefficient between availability and number of reviews is approximately 0.188975, suggesting a weak to moderate positive linear relationship between these variables.

Correlation between Number of Reviews and Ratings:

- Investigated the correlation between the number of reviews and the ratings of Airbnb listings.
- Computed the correlation coefficient, filtering out null values for number of reviews or ratings.
- Result: The correlation coefficient between number of reviews and ratings is approximately -0.002888, indicating almost no linear relationship between these variables.

These analyses provide insights into the relationships between various factors such as price, availability, number of reviews, and ratings of Airbnb listings. The correlation coefficients help understand the strength and direction of these relationships, assisting in making informed decisions related to Airbnb hosting or accommodation choices.